src/datastore/sync_methods/inject.js

```diff
@@ -313,12 +313,10 @@ module.exports = function inject (resourceName, attrs, options) {
     definition.emit('DS.beforeInject', definition, attrs)
   }
 
-  // start the recursive injection of data
+  // start the recursive injection of data; this will also call
+  // `definition.handleChange` for each item
   injected = _inject.call(_this, definition, resource, attrs, options)
 
-  // collection was modified
```

Comment:
The problem is, this introduces a breaking change for anyone currently relying on the current API. Breaking changes require a major version bump. So, you'll need a way to add the feature without a breaking change, which usually involves making it an option.